kv/util/util.go

```diff
@@ -33,54 +34,179 @@ import (
 var errNilStore = errors.New("kv store is nil")
 
 type getValueFn func(kv.Value) (interface{}, error)
+
 type updateFn func(interface{})
 
+type validateBoundsFn func(interface{}) (lower, upper interface{}, err error)
```

Comment:
More generic to have a `validateFn` with signature: 

```
type validateFn func(interface{}) error 
```

It'll be type agnostic and allow for more custom validation if we ever need it